com.unity.ml-agents/Editor/MLAgentsSettingsProvider.cs

```diff
@@ -0,0 +1,187 @@
+                path += ".asset";
+
+            var relativePath = "Assets/" + path.Substring(assetPath.Length);
+            CreateNewSettingsAsset(relativePath);
+        }
+
+        private static void CreateNewSettingsAsset(string relativePath)
+        {
+            var settings = ScriptableObject.CreateInstance<MLAgentsSettings>();
+            AssetDatabase.CreateAsset(settings, relativePath);
+            EditorGUIUtility.PingObject(settings);
+            // Install the settings. This will lead to an MLAgentsManager.OnSettingsChange event
+            // which in turn will cause this Provider to reinitialize
+            MLAgentsManager.Settings = settings;
+        }
+
+        public override void OnGUI(string searchContext)
+        {
+            if (m_Settings == null)
+            {
+                InitializeWithCurrentSettings();
+            }
+
+            if (m_AvailableSettingsAssets.Length == 0)
+            {
+                EditorGUILayout.HelpBox(
+                    "Click the button below to create a settings asset you can edit.",
+                    MessageType.Info);
+                if (GUILayout.Button("Create settings asset", GUILayout.Height(30)))
+                    CreateNewSettingsAsset();
+                GUILayout.Space(20);
+            }
+
+            using (new EditorGUI.DisabledScope(m_AvailableSettingsAssets.Length == 0))
+            {
+                EditorGUI.BeginChangeCheck();
+                EditorGUILayout.PropertyField(m_SettingsObject.FindProperty("m_ConnectTrainer"), new GUIContent("Connect to Trainer"));
+                EditorGUILayout.PropertyField(m_SettingsObject.FindProperty("m_EditorPort"), new GUIContent("Editor Training Port"));
+                if (EditorGUI.EndChangeCheck())
+                    m_SettingsObject.ApplyModifiedProperties();
+            }
+        }
+
+        internal void InitializeWithCurrentSettings()
+        {
+            m_AvailableSettingsAssets = FindSettingsInProject();
+
+            m_Settings = MLAgentsManager.Settings;
+            var currentSettingsPath = AssetDatabase.GetAssetPath(m_Settings);
+            if (string.IsNullOrEmpty(currentSettingsPath))
+            {
+                if (m_AvailableSettingsAssets.Length > 0)
+                {
+                    m_CurrentSelectedSettingsAsset = 0;
+                    m_Settings = AssetDatabase.LoadAssetAtPath<MLAgentsSettings>(m_AvailableSettingsAssets[0]);
+                    MLAgentsManager.Settings = m_Settings;
+                }
+            }
+            else
+            {
+                var settingsList = m_AvailableSettingsAssets.ToList();
+                m_CurrentSelectedSettingsAsset = settingsList.IndexOf(currentSettingsPath);
+
+                EditorBuildSettings.AddConfigObject(MLAgentsManager.EditorBuildSettingsConfigKey, m_Settings, true);
+            }
+
+            m_SettingsObject = new SerializedObject(m_Settings);
```

Comment:
I get an error on this line :
```
ArgumentException: Object at index 0 is null
UnityEditor.SerializedObject..ctor (UnityEngine.Object obj) (at /Users/bokken/buildslave/unity/build/Editor/Mono/SerializedObject.bindings.cs:23)
Unity.MLAgents.Editor.MLAgentsSettingsProvider.InitializeWithCurrentSettings () (at /Users/vincentpierre/Documents/ml-agents/com.unity.ml-agents/Editor/MLAgentsSettingsProvider.cs:170)
```
Seems to happen on both 2018.4 and 2020.1

Reply:
Tested with single settings file and multiple files, and broke the case with no file..
Should be fixed now.